### ISSUE TITLE
Audio extractor fails to extract ME2 AFCs with "Could not find file 'exec\out.wav'"

### ISSUE THREAD

**Issue by GreatEmerald** [ME3Explorer/ME3Explorer#468]
When trying to extract any Mass Effect 2 audio file, the Audio Extractor utility from ME3Explorer 3.0.0 fails and throws an exception:

```
System.IO.FileNotFoundException: Could not find file 'F:\ME2\ME3Explorer\exec\out.wav'.
File name: 'F:\ME2\ME3Explorer\exec\out.wav'
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.File.InternalCopy(String sourceFileName, String destFileName, Boolean overwrite, Boolean checkHost)
   at System.IO.File.Copy(String sourceFileName, String destFileName)
   at ME3Explorer.AFCExtract.allToolStripMenuItem_Click(Object sender, EventArgs e)
   at System.Windows.Forms.ToolStripItem.RaiseEvent(Object key, EventArgs e)
   at System.Windows.Forms.ToolStripMenuItem.OnClick(EventArgs e)
   at System.Windows.Forms.ToolStripItem.HandleClick(EventArgs e)
   at System.Windows.Forms.ToolStripItem.HandleMouseUp(MouseEventArgs e)
   at System.Windows.Forms.ToolStripItem.FireEventInteractive(EventArgs e, ToolStripItemEventType met)
   at System.Windows.Forms.ToolStripItem.FireEvent(EventArgs e, ToolStripItemEventType met)
   at System.Windows.Forms.ToolStrip.OnMouseUp(MouseEventArgs mea)
   at System.Windows.Forms.ToolStripDropDown.OnMouseUp(MouseEventArgs mea)
   at System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.ScrollableControl.WndProc(Message& m)
   at System.Windows.Forms.ToolStrip.WndProc(Message& m)
   at System.Windows.Forms.ToolStripDropDown.WndProc(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.OnMessage(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
```

From what I can tell, this is because `ww2ogg.exe` doesn't process the `out.dat` file properly by default (the resulting OGG file is just a few bytes in size), and requires passing the `--full-setup` parameter to properly extract the audio.

**Comment by ghost** [comment]
The extractor likely hasn't been touched since originally created, and I'm unsure whether or not ME2 support was originally intended. It is functional for ME3 last I checked.

The functionality of this tool is going to be rolled into the new Audio Editor, which will also combine Soundplorer, the Wwisebank Editor, and Fob's external AFC creator. For that reason, I'm going to close your Issue with a "won't fix". Appreciate the heads up, though. Thanks.

**Comment by GreatEmerald** [comment]
Like I said, I tested the manual workflow (running `ww2ogg.exe` on an AFC or part of it, and `oggdec.exe` on the resulting OGG to get a WAV), and it works for ME2 files fine as long as the `--full-setup` parameter to `ww2ogg.exe` is passed. Though `ww2ogg.exe` only handles a single output file, so if passed a whole AFC file, it will only extract the first audio element from it. That's the main reason why the audio extractor is useful to begin with: for finding the needed offsets and cutting the files up.

But yes, I take it you want to fold this into issue #410.

**Comment by ghost** [comment]
@GreatEmerald -- Just wanted to follow up with a few things:

1. Individual sound files for dialogue, music, or whatnot inside AFCs can already be extracted individually with Soundplorer (which only supports ME3, atm). That functionality will be carried over into the new tool. 

2. The functionality of the extractor in the new Audio Editor will be as a supplemental menu feature capable of fully extracting a single AFC, a selection of AFCs, or all AFCs in a chosen directory. This bulk feature is the extractor's strong point and is exactly why it should be carried over into the new tool. I'm guessing this is what you mean by "cutting the files up"?

3. The new Audio Editor will still only support ME3 at first. Most tools in the toolset do not support ME2 -- there are only a small handful. Once we having it working for ME3, then it will be ported to ME2.

4. As far as the goal with the extractor feature of the new editor, it should prevent the player from having to use Kyp's or Gibbed's tools, both which you probably know work for ME2 and ME3. ME1 will be more complicated.

---

I closed the issue since the code for the Audio Extractor as it currently exists should no longer be edited. It's better to make the change in the code for the new tool. If @SirCxyrtyx wants to include this change for ME2 support, it's up to him. He may also prefer to wait until we're officially building in support for ME2 for the entire tool, which won't happen when initially launched.

**Comment by GreatEmerald** [comment]
1) Yea, I looked at Soundplorer, but it doesn't accept ME2 AFC files indeed.
2) I meant multiple audio clips within one AFC file. Extracting all audio clips from all AFCs in the game sounds like way overkill for most purposes...
3) I see.

Well, to be honest, I was just looking for a tool that would let me extract the news bits from ME2, and since I had ME3Explorer, I figured it would have one. It did, but it's not working, hence this bug report ;) I guess I'll have to look for other tools in the mean while (unless the new tool is made and supports ME2 in about half a year or so, I'm in no rush).
